File: coding/08th July 2025/Maximum Difference Between Even and Odd Frequency 1.java
//Maximum Difference Between Even and Odd Frequency 1
class Solution 
{
    public int maxDifference(String s) 
    {
        HashMap<Character, Integer> map = new HashMap<>();
        for(char ch : s.toCharArray())
        {
            if(map.containsKey(ch))
                map.put(ch, map.get(ch) + 1);
            else
                map.put(ch, 1);
        }
        int max = Integer.MIN_VALUE, min = Integer.MAX_VALUE;
        for(char key : map.keySet())
        {
            int val = map.get(key);
            if(val % 2 != 0)
            {
                if(val > max)
                    max = val;
            }
            else if(val % 2 == 0)
            {
                if(val < min)
                    min = val;
            }
        }
        if(max == Integer.MIN_VALUE || min == Integer.MAX_VALUE)
            return -1;
        return max - min;
    }
}

File: Coding in Python/14th January 2026/Maximum Difference Between Even and Odd Frequency 1.py
#Maximum Difference Between Even and Odd Frequency 1
class Solution:
    def maxDifference(self, s: str) -> int:
        d = {}
        mo = float('-inf')
        me = float('inf')
        for i in range(len(s)):
            if s[i] in d:
                d[s[i]] = d.get(s[i]) + 1
            else:
                d[s[i]] = 1
        for val in d.values():
            if val % 2 != 0:
                mo = max(mo, val)
            else:
                me = min(me, val)
        if mo == float('-inf') or me == float('inf'):
            return -1
        return mo - me
